client-core/src/service/key_service.rs

```diff
@@ -1,38 +1,70 @@
 use secstr::SecUtf8;
-use zeroize::Zeroize;
 
-use client_common::{PrivateKey, PublicKey, Result, SecureStorage, Storage};
-
-const KEYSPACE: &str = "core_key";
+use super::basic_key_service::BasicKeyService;
+use super::hdkey_service::HDKeyService;
+use super::key_service_data::{KeyServiceInterface, WalletKinds};
+use client_common::{PrivateKey, PublicKey, Result, Storage};
 
 /// Maintains mapping `public-key -> private-key`
 #[derive(Debug, Default, Clone)]
 pub struct KeyService<T: Storage> {
-    storage: T,
+    kind: WalletKinds,
+    basic: Option<BasicKeyService<T>>,
+    hd: Option<HDKeyService<T>>,
 }
 
 impl<T> KeyService<T>
 where
     T: Storage,
 {
     /// Creates a new instance of key service
-    pub fn new(storage: T) -> Self {
-        KeyService { storage }
+    pub fn new(storage: T, stroage2: T, kind: WalletKinds) -> Self {
+        KeyService {
+            kind,
+            basic: Some(BasicKeyService::new(storage)),
+            hd: Some(HDKeyService::new(stroage2)),
+        }
+    }
+
+    /// get random mnemonic
+    pub fn get_random_mnemonic(&self) -> String {
+        match self.kind {
+            WalletKinds::Basic => String::new(),
+            WalletKinds::HD => self.hd.as_ref().unwrap().get_random_mnemonic(),
+        }
     }
 
+    /// restore from mnemonic
+    pub fn generate_seed(&self, mnemonic: &str, name: &str, passphrase: &SecUtf8) -> Result<()> {
+        match self.kind {
+            WalletKinds::Basic => Ok(()),
```

Comment:
or return an error, as this operation doesn't make sense on "basic"?

Reply:
modified, normal wallet doesn't call this function